### PR TITLE
fix(EnergyFunds): Added FindVehicleFormField to EnergyFunds application

### PR DESCRIPTION
**Pull request by davidkef**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx [hunk 1]
```diff
@@ -107,6 +114,17 @@ const extractDetails = function (
...
+  } else if (
+    isVehicleType<EnergyFundVehicleDetailsWithGrant>(
+      response,
+      'EnergyFundVehicleDetailsWithGrant',
+    )
+  ) {
+    return {
+      ...response,
+      vehicleGrant: 600000,
+      vehicleGrantItemCode: '123456',
```
Comment (jonarnarbriem) [inline libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx:126]: are these values correct?

**Review by jonarnarbriem — APPROVED** [review]

**Comment by jonarnarbriem** [comment]
Approved with one comment, leaving it up to you if you want to make a change or not

File: libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx [hunk 2]
```diff
@@ -63,14 +66,18 @@ const isVehicleType = function <T>(
   )
 }
 
+export const formatIsk = (value: number): string =>
+  value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, '.') + ' kr.'
+
```
Comment (Toti91) [inline libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx:71]: We have a shared `formatCurrency` function that I would prefer you'd use for consistency

File: libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx [hunk 3]
```diff
@@ -305,6 +357,55 @@ export const FindVehicleFormField: FC<React.PropsWithChildren<Props>> = ({
...
+                heading={`${energyDetails.make ?? ''} - ${
+                  energyDetails.permno
+                }`}
+                text={`${energyDetails.color} - ${
+                  energyFundsMessages &&
+                  formatText(
+                    energyFundsMessages.registrationDate,
+                    application,
+                    formatMessage,
+                  )
```
Comment (Toti91) [inline libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx:371]: If `energyFundsMessages` is false, won't the text end up as "color - false"?
I would rather use a ternary operator and fallback to empty string

```ts
energyFundsMessages ? formatText(....) : ''
```

File: libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx [hunk 4]
```diff
@@ -305,6 +357,55 @@ export const FindVehicleFormField: FC<React.PropsWithChildren<Props>> = ({
...
+                                  energyFundsMessages.carNotEligable as MessageDescriptor,
+                                ),
+                          },
+                        )
+                      : energyFundsMessages &&
+                        formatText(
+                          energyFundsMessages.checkboxNotCheckable,
+                          application,
+                          formatMessage,
+                        )) ?? '',
```
Comment (Toti91) [inline libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx:399]: Lot of logic to have in the markup, maybe move this to a util function in another file

**Review by Toti91 — CHANGES_REQUESTED** [review]
Some comments, otherwise looks good!

File: libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx (hunk 2 above)
Comment (davidkef) [reply libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx:71]: Good pointer, will replace with this one

File: libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx (hunk 3 above)
Comment (davidkef) [reply libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx]: Yes, you are correct. Will fix

File: libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx (hunk 4 above)
Comment (davidkef) [reply libs/application/ui-fields/src/lib/FindVehicleFormField/FindVehicleFormField.tsx]: Moved to another file

**Review by Toti91 — APPROVED** [review]

**Pull request merged**